src/zenml/new/steps/log_artifact_metadata.py

```diff
@@ -0,0 +1,120 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Utility functions for logging artifact metadata."""
+
+from typing import Dict, Optional
+
+from zenml.exceptions import StepContextError
+from zenml.metadata.metadata_types import MetadataType
+from zenml.new.steps.step_context import get_step_context
+
+
+def log_artifact_metadata(
+    output_name: Optional[str] = None,
```

Comment:
Actually we do, this is handled by the `StepContext` which will raise a `StepContextError` if you supply a non-existent output name or if there are multiple outputs and you didn't specify a name at all. See the error handling we do below

Reply:
Ah you couldn't have known that, my explanation for the error in the docstring was wrong, thanks for asking!

Reply:
How we handle unnamed artifacts? If second example would be like this:
```python
@step
def artifact_multi_output_metadata_logging_step() -> str,int:
...
```

Reply:
These are by default called `output_0, output_1, ...`

Reply:
Yeah I think users can figure that out since we even have a separate section on output names in the docs. If not, worst case they see an error that says "name X does not exist, only the following names exist (Y, Z, ...)" which should be fairly clear on how to resolve it